<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>The pipeline</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h3>The pipeline</h3>

<p>The goal of this assignment is to analyze read coverage of the HIV genome and look for potential relationships between nucleotide composition and sequencing coverage. coverage.py is a wrapper for all the steps involved in the analysis. It runs in python2.7. One specification was that the pipeline would be run at different thresholds of sequencing quality. Each of the pipeline runs is a subdirectory of the output directory.</p>

<h3>Commands for running the pipeline</h3>

<pre><code>python2.7 coverage.py -1 ../data/test_1.fastq -2 ../data/test_2.fastq -x ../annotation/HXB2.fasta -a ../annotation -p HXB2 -q 10 -o ../output/test &gt; test.log

python2.7 coverage.py -1 ../data/test_1.fastq -2 ../data/test_2.fastq -x ../annotation/HXB2.fasta -a ../annotation -p HXB2 -q 0 -o ../output/HXB2_q0 &gt; q0.log

python2.7 coverage.py -1 ../data/test_1.fastq -2 ../data/test_2.fastq -x ../annotation/HXB2.fasta -a ../annotation -p HXB2 -q 10 -o ../output/HXB2_q10 &gt; q10.log

python2.7 coverage.py -1 ../data/test_1.fastq -2 ../data/test_2.fastq -x ../annotation/HXB2.fasta -a ../annotation -p HXB2 -q 20 -o ../output/HXB2_q20 &gt; q20.log

python2.7 coverage.py -1 ../data/test_1.fastq -2 ../data/test_2.fastq -x ../annotation/HXB2.fasta -a ../annotation -p HXB2 -q 30 -o ../output/HXB2_q30 &gt; q30.log
</code></pre>

<h3>Arguments</h3>

<h5>Basic arguments specified in the assignment</h5>

<pre><code>-1  # Sense fastq file
-2  # Antisense fastq file
-x  # Genome fasta
-q  # Minimum mean base quality

</code></pre>

<p>Currently the pipeline supports only paired end sequencing and the input files must be well formed, with reads in the same order.</p>

<h5>Additional flags</h5>

<pre><code>-p  # A user supplied prefix for naming the output
</code></pre>

<p>This can be any string that is valid as a director name</p>

<pre><code>-o  # Output directory. This is the destination of the output files
-a  # Annotation directory. This is where the bowtie2 index files are sent
-d  # Data directory. This is the destination of intermediate data files
</code></pre>

<p>By default all of these are set to the current working directory. The user can specify a more meaningful project layout if desired. I kept my genome files in a separate annotation directory and all of my fastq and alignment files in a separate data directory. My output, specified by the assignment is sent to its own directory. None of this is required for the pipeline though.</p>

</body>

</html>

